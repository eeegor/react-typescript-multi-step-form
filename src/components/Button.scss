@import '../variables';

.button {
	font-size: 0.9rem;
	letter-spacing: 0.125rem;
	box-sizing: border-box;
	display: flex;
	padding: 0.85rem 1.5rem 0.75rem;
	border: 0;
	outline: 0;
	background: #ccc;
	border-radius: 4px;
	justify-content: center;
	align-items: center;
	text-transform: uppercase;

	&--large {
		padding: 1.3rem 2.5rem 1.2rem;
		font-size: 1rem;
	}
}

.button--disabled,
input[type='button' i]:disabled,
input[type='submit' i]:disabled,
input[type='reset' i]:disabled,
button:disabled {
	color: #333;
	opacity: 0.3;
}

.button--success {
	color: #fff;
	background-color: $color-success;
}
