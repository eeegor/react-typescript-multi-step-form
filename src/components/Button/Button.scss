@import '../../variables';

.button {
	font-size: 0.9rem;
	letter-spacing: 0.125rem;
	box-sizing: border-box;
	display: flex;
	padding: 0.85rem 1.5rem 0.75rem;
	border: 0;
	outline: 0;
	background: $color-border;
	border-radius: 4px;
	justify-content: center;
	align-items: center;
	text-transform: uppercase;

	&--large {
		padding: 1.3rem 2.5rem 1.2rem;
		font-size: 1rem;
	}
}

.button--disabled,
input[type='button' i]:disabled,
input[type='submit' i]:disabled,
input[type='reset' i]:disabled,
button:disabled {
	color: $color-grey-dark;
	opacity: 0.3;
}

.button--success {
	color: white;
	background-color: $color-success;
}
