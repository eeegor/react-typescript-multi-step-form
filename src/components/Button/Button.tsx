/**
 * Component Button
 */

import React from 'react';
import classnames from 'classnames';
import './Button.scss';

export interface Props {
	size?: 'large' | 'small';
	type?: 'primary' | 'secondary' | 'success' | 'danger' | 'disabled' | 'muted';
	className?: string;
	children: string | JSX.Element;
	onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
}

export const Button = (props: Props): JSX.Element => {
	const { size, className, type, children, onClick } = props;
	const isDisabled = type === 'disabled';
	const cssClasses = classnames(
		'button',
		type && `button--${type}`,
		size && `button--${size}`,
		isDisabled && `button--disabled`,
		className
	);

	if (isDisabled) {
		return (
			<button type="button" disabled={true} className={cssClasses}>
				{children}
			</button>
		);
	}

	return (
		<button
			type="button"
			className={cssClasses}
			onClick={event => onClick && onClick(event)}
		>
			{children}
		</button>
	);
};
