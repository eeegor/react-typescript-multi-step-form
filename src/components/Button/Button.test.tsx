import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import { Button } from './Button';

describe('Button', () => {
	it('renders markup correctly', () => {
		const wrapper = <Button>Hello</Button>;
		const tree = renderer.create(wrapper).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should adopt a disabled attribute and class', () => {
		const wrapper = shallow(<Button type="disabled">Hello</Button>);
		expect(wrapper.prop('disabled')).toEqual(true);
		expect(wrapper.hasClass('button--disabled')).toEqual(true);
	});

	it('should generate a small button', () => {
		const wrapper = shallow(<Button size="small">Hello</Button>);
		expect(wrapper.hasClass('button--small')).toEqual(true);
	});

	it('should generate a large button', () => {
		const wrapper = shallow(<Button size="large">Hello</Button>);
		expect(wrapper.hasClass('button--large')).toEqual(true);
	});

	it('click event works', () => {
		const callback = jest.fn();
		const wrapper = mount(<Button onClick={callback}>Hi</Button>);
		wrapper.find('button').simulate('click');
		expect(callback).toHaveBeenCalledTimes(1);
	});
});
