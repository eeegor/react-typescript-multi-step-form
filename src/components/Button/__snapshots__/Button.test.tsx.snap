// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders markup correctly 1`] = `
<button
  className="button"
  onClick={[Function]}
  type="button"
>
  Hello
</button>
`;
